js/packages/openinference-instrumentation-beeai/src/helpers/getSerializedObjectSafe.ts

```diff
@@ -0,0 +1,325 @@
+  ) {
+    if (!dataObject) {
+      return {
+        [SemanticConventions.OPENINFERENCE_SPAN_KIND]:
+          OpenInferenceSpanKind.TOOL,
+      };
+    }
+    const { input, output, error } = dataObject as InferCallbackValue<
+      | ToolEvents["start"]
+      | ToolEvents["success"]
+      | ToolEvents["retry"]
+      | ToolEvents["error"]
+    >;
+
+    return {
+      [SemanticConventions.OPENINFERENCE_SPAN_KIND]: OpenInferenceSpanKind.TOOL,
+      ...(input
+        ? { [SemanticConventions.TOOL_PARAMETERS]: JSON.stringify(input) }
+        : {}),
+      ...(output && {
+        [SemanticConventions.OUTPUT_VALUE]: JSON.stringify(output),
+      }),
+      ...(error && {
+        "exception.message": error.message,
+        "exception.stacktrace": error.stack,
+        "exception.type": error.name,
+      }),
+    };
+  }
+
+  // llm events
+  if (
+    [
+      successLLMEventName,
+      startLLMEventName,
+      errorLLMEventName,
+      newTokenLLMEventName,
+    ].includes(meta.name as keyof ChatModelEvents) &&
+    meta.creator instanceof ChatModel
+  ) {
+    const { value, input, error } = dataObject as InferCallbackValue<
+      | ChatModelEvents["success"]
+      | ChatModelEvents["start"]
+      | ChatModelEvents["error"]
+      | ChatModelEvents["newToken"]
+    >;
+
+    const creator = meta.creator.createSnapshot();
+    return {
+      [SemanticConventions.OPENINFERENCE_SPAN_KIND]: OpenInferenceSpanKind.LLM,
+      ...(value?.usage?.completionTokens && {
+        [SemanticConventions.LLM_TOKEN_COUNT_COMPLETION]:
+          value?.usage?.completionTokens,
+      }),
+      ...(value?.usage?.promptTokens && {
+        [SemanticConventions.LLM_TOKEN_COUNT_PROMPT]:
+          value?.usage?.promptTokens,
+      }),
+      ...(value?.usage?.totalTokens && {
+        [SemanticConventions.LLM_TOKEN_COUNT_TOTAL]: value?.usage?.totalTokens,
+      }),
+      ...(input?.messages.length > 0 && {
+        [SemanticConventions.INPUT_MIME_TYPE]: MimeType.JSON,
+        [SemanticConventions.INPUT_VALUE]: JSON.stringify(input.messages),
+        ...parserLLMInputMessages(input.messages),
+      }),
+      ...(value?.messages.length > 0 && {
+        [SemanticConventions.OUTPUT_MIME_TYPE]: MimeType.JSON,
+        [SemanticConventions.OUTPUT_VALUE]: JSON.stringify(value.messages),
+        ...parseLLMOutputMessages(value.messages),
+      }),
+      ...(error && {
+        "exception.message": error.message,
+        "exception.stacktrace": error.stack,
+        "exception.type": error.name,
+      }),
+      ...("providerId" in creator && {
+        [SemanticConventions.LLM_PROVIDER]: creator.providerId,
+        [`${SemanticAttributePrefixes.metadata}.ls_${LLMAttributePostfixes.provider}`]:
+          creator.providerId,
+      }),
+      ...("modelId" in creator && {
+        [SemanticConventions.LLM_MODEL_NAME]: creator.modelId,
+        [`${SemanticAttributePrefixes.metadata}.ls_${LLMAttributePostfixes.model_name}`]:
```

Comment:
this seems to be maybe similar to langsmith's metadata paradigm. with openinference we already have these as semantic attributes. Is this necessary for a particular usecase?

Reply:
I am sorry, but I don't understand your suggestion. Could you please provide me with more information about using metadata in this case?
I need to send this model name metadata attribute to correctly show them in the UI.

Reply:
Yes, you've captured the model name and provider correctly already but it looks like you might be also adding langsmith params? https://github.com/langchain-ai/langchain/blob/49bdd3b6fe07e1ee29c5f2381ef7a563bef15778/libs/core/langchain_core/language_models/base.py#L37

Reply:
e.g. you are free to add metadata but the `ls` prefix is unnecessary here.